br/pkg/lightning/restore/table_restore.go

```diff
@@ -1012,3 +1019,11 @@ func estimateCompactionThreshold(cp *checkpoints.TableCheckpoint, factor int64)
 
 	return threshold
 }
+
+func (tr *TableRestore) allocateRowIDs() (int64, int64) {
+	tr.rowIDLock.Lock()
+	defer tr.rowIDLock.Unlock()
+	newBase := tr.curMaxRowID + 1
+	tr.curMaxRowID += defaultNewAllocIDCnt
+	return newBase, tr.curMaxRowID
```

Comment:
We should also consider parallel import. Please refer https://github.com/pingcap/tidb/blob/d8be8edbe11c5602c40c96fc8254b6ab13c035c8/br/pkg/lightning/restore/meta_manager.go#L80.

Reply:
Does parallel import here mean restoring tables from different lightning instances so I'm supposed to read pre-allocated ID from checkpoints before allocating them?

Reply:
In parallel import, we should ensure different instances use different row id.